File: fib.go
package main

import "fmt"

func Fib(n int) int{
	var f int
	var p1 int
	var p2 int
	f = 1
	p1 = 1

	for i:=2;i<n;i++{
		p2 = p1          //   一つ前の結果を二つ前の結果 (p2) へ移す．
		p1 = f           //   直前の結果を一つ前の結果に移す．
		f = p1 + p2
	}
	return f

}

func main() {
	fmt.Println(Fib(10))
	fmt.Println(Fib(50))
	fmt.Println(Fib(100))
}

File: fib.py
import sys


def Fib(n):
    if n == 0:
        return 0
    elif n == 1 or n == 2:
        return 1
    else:
        return Fib(n-1) + Fib(n-2)


def fibnormal(n):
    f = 1 #1
    x1 = 1 #1
    for i in range(2, n): #3*n
        x2 = x1
        x1 = f
        f = x1 + x2
    return f


if __name__=='__main__':
    sys.setrecursionlimit(9000)
    #print(Fib(8181))
    print(fibnormal(10))
    print(fibnormal(50))
    print(fibnormal(100))
    print(fibnormal(8181))
